file-attribute-caching/src/test/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemTests.kt

```diff
@@ -31,10 +34,46 @@ import java.nio.file.attribute.UserPrincipal
 import java.text.SimpleDateFormat
 import java.util.EnumSet
 import java.util.stream.Stream
+import kotlin.concurrent.thread
 import kotlin.io.path.exists
 
 class AttributeCachingFileSystemTests {
 
+    @Disabled // Enable this test if you wish to see how the attribute caching fileSystem fails in a multi-threaded
+    // environment. This test will be very in-consistent but it will throw a ClassCastException when it does fail
+    // (you may have to run it multiple times)
+    @Test
+    fun `scratch test`() {
+        var caughtException: Exception? = null
+
+        for (i in 1..100_000) {
+            thread {
+                try {
+                    AttributeCachingFileSystem.wrapping(FileSystems.getDefault()).use { cachingFilesystem ->
+                        cachingFilesystem.getPath("file.txt")
+                    }
+                    Thread.sleep(1_000)
+                } catch (e: Exception) {
+                    caughtException = e
+                }
+            }
+        }
+
+        thread {
+            try {
+                AttributeCachingFileSystem.wrapping(FileSystems.getDefault()).use { cachingFilesystem ->
+                    cachingFilesystem.getPath("file.txt")
+                    Thread.sleep(10_000)
+                }
+            } catch (e: Exception) {
+                caughtException = e
+            }
```

Comment:
```suggestion
            for (i in 1..100_000) {
                try {
                    AttributeCachingFileSystem.wrapping(FileSystems.getDefault()).use { }
                } catch (e: Exception) {
                    caughtException = e
                }
            }
```

This section could probably also loop to increase the likelihood of collisions.